fuel-vm/src/interpreter/metadata.rs

```diff
@@ -131,6 +135,13 @@ pub(crate) fn metadata(
             Some(p) => (p == 0) as Word,
             None => return Err(PanicReason::ExpectedInternalContext.into()),
         },
+        GMArgs::GetGasPrice => match context {
```

Comment:
Hmm, it looks like storing `gas_price` in the context would have more sense, and it automatically will return error in the case of predicate because you just don't have a place where you can get it=)